### DISCUSSION TITLE
how inject custom resource proivder

### DISCUSSION THREAD

**Discussion by krishnaexplore** [open-telemetry/opentelemetry-java-instrumentation#10979] [Q&A]
we have resource attributes providers [here](https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/release/v2.1.x/instrumentation/resources/library/src/main/java/io/opentelemetry/instrumentation/resources), how to inject our own custom resource attribute provider

**Comment by laurit** [comment]
You can package your own resource provider as agent extension https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/examples/extension
